### PR TITLE
Remove support for data plane config from environment

### PR DESCRIPTION
**Pull request by lfarrel6**
# Why
The data plane supported both environment and file based configuration to be backwards compatible. Moving towards v1, this is no longer required and we can migrate to exclusively using the config file.

# How
- Remove any code which fell back to environment variable config
- Remove all code for reading from env vars
- Update the feature context to return an error instead of panicking internally

### CODE REVIEW

File: data-plane/src/lib.rs [hunk 1]
```diff
@@ -154,71 +156,47 @@ impl From<ProvisionerContext> for CageContext {
     }
 }
 
+#[derive(Clone, Deserialize)]
+pub struct FeatureContext {
+    pub api_key_auth: bool,
+    pub healthcheck: Option<String>,
+    pub trx_logging_enabled: bool,
+    #[serde(default)]
```
Comment (hanneary) [inline data-plane/src/lib.rs:164]: Were these defaults a backwards compat thing? just noticed trx_logging_enabled doesn't have it

**Review by hanneary — APPROVED** [review]
LGTM just one q

File: data-plane/src/lib.rs (hunk 1 above)
Comment (lfarrel6) [reply data-plane/src/lib.rs:164]: Yep probably were, can remove them assuming they're no longer optional starting v1 of the CLI

Comment (lfarrel6) [reply data-plane/src/lib.rs:164]: discussed offline — v1 of the CLI will make all of these fields required, so defaults can be removed.

**Pull request merged**